core/sql/executor/JavaObjectInterface.cpp

```diff
@@ -147,19 +147,29 @@ int JavaObjectInterface::createJVM()
     }
   }
 
-#ifdef _DEBUG
   int debugPort = 0;
   const char *debugPortStr = getenv("JVM_DEBUG_PORT");
   if (debugPortStr != NULL)
      debugPort = atoi(debugPortStr);
-  const char *debugTimeoutStr = getenv("JVM_DEBUG_TIMEOUT");
-  if (debugTimeoutStr != NULL)
-     debugTimeout_ = atoi(debugTimeoutStr);
-  const char *suspendOnDebug = getenv("JVM_SUSPEND_ON_DEBUG");
-  char debugOptions[300];
-  if (debugPort > 0)
+  if (debugPort > 0
+#ifdef NDEBUG
+      // in a release build, only DB__ROOT can debug a process
+      && ComUser::isRootUserID()
+#endif
+     )
   {
-     debugPort_ = debugPort + (GetCliGlobals()->myPin() % 1000);
+     const char *debugTimeoutStr = getenv("JVM_DEBUG_TIMEOUT");
+     if (debugTimeoutStr != NULL)
+       debugTimeout_ = atoi(debugTimeoutStr);
+     const char *suspendOnDebug = getenv("JVM_SUSPEND_ON_DEBUG");
+     char debugOptions[300];
+
+     debugPort_ = debugPort;
+     // to allow debugging multiple processes at the same time,
+     // specify a port that is a multiple of 1000 and the code will
+     // add pid mod 1000 to the port number to use
+     if (debugPort_ % 1000 == 0)
+       debugPort_ += (GetCliGlobals()->myPin() % 1000);
```

Comment:
What's the probability of port collisions? How does one identify them? I suppose a way to work around port collisions is to stop and restart the affected processes?